### DISCUSSION TITLE
String literal concatenation

### DISCUSSION THREAD

**Discussion by mcejp** [hylang/hy#2476] [Q&A] [answered]
In Python, multiple adjacent string or bytes literals (delimited by whitespace), possibly using different quoting conventions, are allowed, and their meaning is the same as their concatenation ([source](https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation)). 

I was wondering if an equivalent construct exists in Hy, or more generally, what would be the recommended way of splitting f-strings that are too long to fit on a single line.

Merci :wink:

**Comment by scauligi** [comment, marked as answer]
Since Hy has prefix operators, it's easiest to just explicitly concatenate your different strings. Indeed, the `pprint` function from Hyrule does exactly that:

```hy
=> (hyrule.pprint "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.")
(+ "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod "
   "tempor incididunt ut labore et dolore magna aliqua.")
```

**Reply by Kodiologist** [reply]
I always thought Python's implicit string-literal concatenation was a weird holdover from C. Dangerous, too: forget the comma in `["foo", "bar"]` and you get something different with no warning.
